wallet/src/repository/wallet/mod.rs

```diff
@@ -315,38 +316,19 @@ where
         txns: &[types::VTTransactionBody],
     ) -> Result<()> {
         // FIXME: Handle multiple accounts when indexing transactions
-        let account = 0;
         let mut state = self.state.write()?;
 
         for txn in txns {
             let hash = txn.hash().as_ref().to_vec();
-
-            match self.db.get_opt(&keys::transactions_index(&hash))? {
-                Some(txn_id) => {
-                    self.db
-                        .put(&keys::transaction_block(account, txn_id), block)?;
-                }
-                None => {
-                    for input in &txn.inputs {
-                        self._index_transaction_input(&mut state, &hash, &input, block)?;
-                    }
-
-                    for (index, output) in txn.outputs.iter().enumerate() {
-                        let out_ptr = model::OutPtr {
-                            txn_hash: hash.clone(),
-                            output_index: index as u32,
-                        };
-                        self._index_transaction_output(&mut state, &hash, out_ptr, &output, block)?;
-                    }
-                }
+            if self
+                .db
+                .get_opt::<_, u32>(&keys::transactions_index(&hash))?
+                .is_none()
```

Comment:
Add comments about why are we checking that this value is `None`, and explain what happens when the value is `Some` (it is ignored?)

Reply:
If `Some` we will log a warning because the transaction is already in our database. This should never happen. :man_facepalming: